### PR TITLE
Feat/track details page

### PR DESCRIPTION
**Pull request by adidoshiws**
### Ticket Link

---

### Related Links

---

### Description

- Created track details page
- When clicked on the trackCard it redirects to the different route `/tracks/trackId`
- updated saga, reducers, selectors

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---
containerPRDemo7

### Live Link

---

### CODE REVIEW

File: app/containers/TrackDetailsContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+const TrackImage = styled(Image)`
+  width: ${(props) => props.width}px;
+`;
+
+const PlayBtn = styled(Button)`
+  type: ${(props) => props.type};
+`;
+
+export function TrackDetailsContainer({ dispatchTrackDetails, trackDetailsData, trackDetailsError, loading }) {
```
Comment (tousifws) [inline app/containers/TrackDetailsContainer/index.js]: Isn't `loading` undefined? It's not getting `loading` state from redux and also not defined in reducer. Handle the loading state from reducer in ItunesContainer also.

Comment (adidoshiws) [reply app/containers/TrackDetailsContainer/index.js]: yeah, updated the loading state!

File: app/containers/ItunesContainer/index.js [hunk 2]
```diff
@@ -54,33 +54,27 @@ export function ItunesContainer({
...
   padding
 }) {
-  const [loading, setLoading] = useState(false);
-
   useEffect(() => {
-    const loaded = get(tracksData, 'results', null) || tracksError;
-    if (loaded) {
-      setLoading(false);
-    }
+    get(tracksData, 'results', null) || tracksError;
```
Comment (tousifws) [inline app/containers/ItunesContainer/index.js]: What is this for?

Comment (adidoshiws) [reply app/containers/ItunesContainer/index.js]: That was not needed, removing it.

File: app/routeConfig.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ import NotFound from '@containers/NotFoundPage/Loadable';
 import HomeContainer from '@containers/HomeContainer/Loadable';
 import routeConstants from '@utils/routeConstants';
 import ItunesContainer from '@containers/ItunesContainer/Loadable';
+// import TrackDetails from '@components/TrackDetails/index';
```
Comment (tousifws) [inline app/routeConfig.js]: Remove this commented out code. Otherwise looks good

**Review by tousifws — APPROVED** [review]

File: app/routeConfig.js (hunk 3 above)
Comment (adidoshiws) [reply app/routeConfig.js]: yes sure!

**Pull request merged**